### PR TITLE
Set securityContext on injected initContainer

### LINKED ISSUE #1058: Java Instrumentation with runAsNonRoot
I added `instrumentation.opentelemetry.io/inject-java: "my-instrumentation"` to a pod with `runAsNonRoot` enabled. I get this error: `Error: container has runAsNonRoot and image will run as root (pod: "mypod-58f5d96bbc-tq77x_ns(2eb77116-d336-41ec-a064-eea6b6be10c0)", container: opentelemetry-auto-instrumentation)`. Removing `runAsNonRoot` fixes the issue, but that's not a proper fix. I guess the injected container should copy the `securityContext` from the main parent?

### LINKED ISSUE #1084: InitContainer lacks allowPrivilegeEscalation setting which triggers Azure Gatekeeper OPA
Hello,

I'm afraid to report that OpenTel instrumentation cannot be used with Azure Gatekeeper OPA because of Privilege Escalation not being explicitly set to False on the instrumentation initContainer securityContext.

image

Unless I'm mistaken I cant seem to find a way to explicitly override this. Would it be possible to set this (if privilege escalation is required) or allow it to be overridden via YAML Values?

### PR DESCRIPTION
**Pull request by addreas**
Encountered the issue where the injected `initContainer` was rejected because of its securityContext. This PR simply takes the securityContext from the `container` that is targeted and uses that for the `initContainer` as well.

resolves #1084 
resolves #1058

### CODE REVIEW

**Review by pavolloffay — COMMENTED** [review]
could you please add a test for this? Either unit or e2e

**Comment by pavolloffay** [comment]
Also the CI is failing 🙃

**Comment by addreas** [comment]
Updated the code so that it doesn't break when there isn't any `securityContext` set on the container, and extended the e2e tests to also check for this change.

Props for a nice test setup!

**Review by yuriolisa — APPROVED** [review]
LGTM.

**Review by frzifus — APPROVED** [review]
thanks @addreas !

**Comment by addreas** [comment]
Rebased and ready for merge AFAIK

**Comment by ranga543** [comment]
@frzifus Any update on this one, Seems like to ready to merge ?

**Comment by addreas** [comment]
Rebased on main and added changelog entry. Hadn't realized that was a thing before.

**Comment by santhotech** [comment]
Hi, We are blocked by this as well, any chance we can get this PR approved and merged with the next version?

**Comment by marcoboffi** [comment]
+1

**Review by iblancasa — APPROVED** [review]

**Comment by frzifus** [comment]
ping @open-telemetry/operator-maintainers

File: pkg/instrumentation/sdk.go [hunk 1]
```diff
@@ -149,6 +149,19 @@ func (i *sdkInjector) inject(ctx context.Context, insts languageInstrumentations
 		pod = i.injectCommonEnvVar(otelinst, pod, index)
 		pod = i.injectCommonSDKConfig(ctx, otelinst, ns, pod, index, index)
 	}
+
+	pod = i.setInitContainerSecurityContext(pod, pod.Spec.Containers[index].SecurityContext)
+
+	return pod
+}
+
+func (i *sdkInjector) setInitContainerSecurityContext(pod corev1.Pod, securityContext *corev1.SecurityContext) corev1.Pod {
```
Comment (jaronoff97) [inline pkg/instrumentation/sdk.go:158]: thought: in the future we may want to rename this to be more generic for other container features we want propagated.

**Review by jaronoff97 — APPROVED** [review]

**Comment by marcoboffi** [comment]
In which milestone will it be included?

**Pull request merged**